src/Products/ZCatalog/Catalog.py

```diff
@@ -907,8 +907,11 @@ def _sort_nbest_reverse(self, actual_result_count, result, rs,
                     # This document is not in the sort key index, skip it.
                     actual_result_count -= 1
                 else:
-                    if n >= limit and key >= best:
-                        continue
+                    try:
```

Comment:
Similar code is present in the companion `_sort_nbest`, too. If the change is necessary here, almost surely, it will be necessary there as well.